jsonschema_validation: Group errors by type

odscjames opened the pull request:
https://github.com/Open-Telecoms-Data/cove-ofds/issues/4 https://github.com/Open-Telecoms-Data/cove-ofds/issues/22

jarofgreen committed:
jsonschema_validation: Group errors by type

https://github.com/Open-Telecoms-Data/cove-ofds/issues/4
https://github.com/Open-Telecoms-Data/cove-ofds/issues/22

ghost commented:
Would you 2 like to try this and see if it's ok (where ok == better than what's there!)

It isn't complete - it doesn't have all the right columns on the tables yet. But it does break up that panel by type and add the titles and description you wanted, and it improves the path.

duncandewhurst commented:
Looks good! A few things to update (aside from the table columns):

* Add quotes to string values, for example, in the image below, 'yes' should be enclosed in double quotes because it is a string value. Otherwise, it would be impossible to distinguish between `true` and `"true"`.
* Render words in backticks as code, for example, in the image below, \`true\`, \`false\` and \`"\` should be rendered as `true`, `false` and `"` 
* Render markdown links as HTML links
![image](https://user-images.githubusercontent.com/19265814/200957208-08c1f788-ee3b-4ab5-9ebc-25c50781e30a.png)

jarofgreen committed:
jsonschema_validation: Markdown links to html links

https://github.com/Open-Telecoms-Data/cove-ofds/pull/32#issuecomment-1309483332
https://github.com/Open-Telecoms-Data/cove-ofds/issues/4
https://github.com/Open-Telecoms-Data/cove-ofds/issues/22

ghost commented:
Thanks!

> Add quotes to string values,

Moved to https://github.com/Open-Telecoms-Data/cove-ofds/issues/22#issuecomment-1309907233

> Render words in backticks as code

Moved to https://github.com/Open-Telecoms-Data/cove-ofds/issues/35

> Render markdown links as HTML links

Got 1, missed 2 - added second commit to fix those.

Over to you @Lathrisk

jarofgreen committed:
jsonschema_validation: Add comments

https://github.com/Open-Telecoms-Data/cove-ofds/issues/4
https://github.com/Open-Telecoms-Data/cove-ofds/issues/22

jarofgreen committed:
process: Add download_geojson_meta_url var from ConvertJSONIntoGeoJSON

Lathrisk commented on cove_ofds/templates/cove_ofds/jsonschema_validation_panel.html:
```diff
@@ -0,0 +1,151 @@
+
+{% if 'prefixItems' in validation_errors %}
+    <h4>{% trans 'prefixItems' %}</h4>
+    <p>DESCIRPTION</p>
```
Wee typo here (i.e. `DESCRIPTION` vs `DESCIRPTION`)

odscjames replied:
It's a placeholder anyway as the source spreadsheet was blank - I'll open a issue to point @duncandewhurst to it
